File: Day2/Part1.cpp

file = "D:\\Coding\\C++\\adventofcodeprog\\adventofcode\\adventofcode\\day2.txt"

def main(line):

    colourNum = 0
    gameDoable = True
    x = line.split(":")
    x.pop(0)
    newx = x
    newx = x[0].split(";")
    for section in newx:
        whitespacesplit = section.split()
        correctColours = {"blue": 14, "green": 13, "red": 12}

        for individualValue in whitespacesplit:

            if "," in individualValue:
                individualValue = individualValue.replace(',', '')

            if individualValue in correctColours:

                correctColours[individualValue] = correctColours[individualValue] - colourNum
                if correctColours[individualValue] < 0:
                    gameDoable = False
            else:
                if individualValue.isnumeric():
                    colourNum = int(individualValue)
                else:
                    print(individualValue)
        #print(correctColours)
    return gameDoable


if __name__ == '__main__':
    print("reading file")
    totalTrueGames = 0
    txtFile = open(file, "r")
    lineNum = 1
    for lines in txtFile:
        if main(lines):
            totalTrueGames += lineNum
        lineNum += 1

    print(totalTrueGames)

File: Day4/Part1.py
import math
file = "magic"

totalTrueGames = 0

# The less brackets the greater the bit wizard

def main(line):
    winningNumbersDict = {}
    x = line.split(":")
    x.pop(0)
    totWinningNumbers = 0
    winningNumbersAssigned = False
    newx = x
    newx = x[0].split("|")
    for section in newx:
        whitespacesplit = section.split()
        if not winningNumbersAssigned:
            winningNumbersAssigned = True
            for individualValue in whitespacesplit:
                winningNumbersDict[individualValue] = 0
        else:
            for individualValue in whitespacesplit:
                if individualValue in winningNumbersDict:
                    totWinningNumbers += 1

    return totWinningNumbers

if __name__ == '__main__':
    print("Reading File")
    txtFile = open(file, "r")
    totalPoints = 0

    for lines in txtFile:
        totalTrueGames = main(lines)

        if totalTrueGames != 0:
            powerOf = totalTrueGames - 1
            totalPoints = totalPoints + math.pow(2,powerOf)
            print(totalPoints)

        # 2^n-1
